corehq/ex-submodules/casexml/apps/phone/tests/test_dbaccessors.py

```diff
@@ -44,3 +44,13 @@ def test_get_sync_logs_for_user(self):
     def test_get_last_synclog_for_user(self):
         self.assert_docs_equal(
             get_last_synclog_for_user(self.user_id), self.sync_logs[0])
+
+    def get_and_save_legacy_synclog(self):
+        legacy_sync_log = self.legacy_sync_logs[0]
+        get_db(None).put_attachment(legacy_sync_log._doc, 'test', 'test_attach', 'text/plain')
+
+        sync_log = get_properly_wrapped_sync_log(legacy_sync_log._id)
+        self.assertIn('test_attach', sync_log._attachments)
+
+        # this used to fail for docs with attachments
+        sync_log.save()
```

Comment:
Still confused as to why this works for docs with no attachments before we removed the rev. @dannyroberts do you know?
```python
legacy_sync_log = self.legacy_sync_logs[0]

sync_log = get_properly_wrapped_sync_log(legacy_sync_log._id)

print sync_log._rev
# > 1-bb1e1fa39045409951373580f11b4b9f
sync_log.save()  # this works even though the doc has a rev (as long as it doesn't have attachments)
```